### PR TITLE
Clean up browser-based testing

### PR DESCRIPTION
**Pull request by rohanjr**
- Select elements using more specific CSS class names
- Close the browser page (seems to fix what I thought was a Puppeteer
issue)
- Abstract out the party names slightly and leave comment about party
management service
- Remove use of `waitForSelector` which I think is unnecessary

### CODE REVIEW

File: ui/src/components/LoginScreen.tsx [hunk 1]
```diff
@@ -84,9 +84,14 @@ const LoginScreen: React.FC<Props> = ({onLogin}) => {
                   iconPosition='left'
                   placeholder='Username'
                   value={username}
+                  className='test-select-username-field'
```
Comment (cocreature) [inline ui/src/components/LoginScreen.tsx]: Hm, it seems slightly annoying that this means that the test code will leak into what you actually ship to users. I wonder how people usually handle this. Can you disable those classes in production mode or something like that? Maybe @garyverhaegen-da knows something?

Either way, happy to leave this for a separate PR.

Comment (garyverhaegen-da) [reply ui/src/components/LoginScreen.tsx]: I don't think that's an issue at all. First, it's only a few bytes, but much more importantly, it allows you to run (at least some of) your tests against the live production website, which is always handy.

File: ui/src/index.test.tsx [hunk 2]
```diff
@@ -69,16 +69,17 @@ afterAll(async () => {
...
-  // are left running even after the browser is closed.
-  // See https://github.com/puppeteer/puppeteer/issues/1825.
   if (browser) {
     browser.close();
   }
 });
 
+// Note(cocreature): Once the party management service is exposed via the HTTP JSON API,
+// I would recommend to use the party management service to allocate parties. If you don’t
+// supply a party id hint you will always get a fresh party.
```
Comment (hurryabit) [inline ui/src/index.test.tsx]: Until we have the party allocation service, we could add a "function" which returns a fresh party each time you call it, like `P1, `P2`, ...

Comment (rohanjr) [reply ui/src/index.test.tsx]: Good idea

Comment (rohanjr) [reply ui/src/index.test.tsx]: This is addressed in https://github.com/digital-asset/create-daml-app/pull/69/commits/86ee9b330a2635faaeb49ced40d6a97fec893054
